iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/SubstationImpl.java

```diff
@@ -155,7 +156,9 @@ public Substation addGeographicalTag(String tag) {
         if (tag == null) {
             throw new ValidationException(this, "geographical tag is null");
         }
+        Set<String> oldValue = geographicalTags.stream().map(String::new).collect(Collectors.toSet());
         geographicalTags.add(tag);
```

Comment:
I think you should take into account the return of the `Set::add` and do not notify if the new tag already exists.

I don't know what is the more efficient: doing a copy using the constructor or doing a copy using stream:
```
Set<String> oldValue = new LinkedHashSet<>(geograpicalTags);
```